### PR TITLE
test-getboolfromserviceannotation [cloud provider openstack]

### PR DESCRIPTION
**Pull request by FavourEva**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/openstack/loadbalancer_test.go [hunk 1]
```diff
@@ -534,3 +535,149 @@ func Test_getListenerProtocol(t *testing.T) {
...
+			name: "Valid 'true' annotation with default 'false'",
+			service: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{"annotation7": "true"},
+				},
+			},
+			annotationKey:  "annotation7",
+			defaultSetting: false,
+			expect:         true,
+		},
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:635]: This is the same as the `Valid 'true' annotation` case, isn't it?

File: pkg/openstack/loadbalancer_test.go [hunk 2]
```diff
@@ -534,3 +535,149 @@ func Test_getListenerProtocol(t *testing.T) {
...
+			name: "Valid 'false' annotation with default 'true'",
+			service: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{"annotation8": "false"},
+				},
+			},
+			annotationKey:  "annotation8",
+			defaultSetting: true,
+			expect:         false,
+		},
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:646]: And this one seems the same as `Valid 'false' annotation` if I see correctly?

**Review by dulek — CHANGES_REQUESTED** [review]
Seems like some cases are duplicated if I read this correctly. Can you confirm and remove the duplicates?

**Comment by dulek** [comment]
CI also tells us that code is not formatted with `gofmt`: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2397/pull-cloud-provider-openstack-check/1710300693216104448#1:build-log.txt%3A180

File: pkg/openstack/loadbalancer_test.go (hunk 1 above)
Comment (FavourEva) [reply pkg/openstack/loadbalancer_test.go:635]: They cover similar grounds but the default settings differs. I thought of keeping them separate to maintain district test for each combination of annotation and default setting.

I will immediately merge them into single cases that tests different default settings for the same annotation value to reduce redundancy in the test cases

Is this ok?

Comment (dulek) [reply pkg/openstack/loadbalancer_test.go:635]: `Valid 'true' annotation` has:
```
Annotations: map[string]string{"annotation2": "false"},
defaultSetting: false,
expect:         true,
```

`"Valid 'true' annotation with default 'false'"` has:
Annotations: map[string]string{"annotation7": "true"},
defaultSetting: false,
expect:         true,

The `defaultSetting` is `false` in both. They still feel redundant to me, so yeah, let's make sure we don't have two tests for the exactly the same case. Thank you!

Comment (FavourEva) [reply pkg/openstack/loadbalancer_test.go]: Sorry about. Confirmed and resolved it.
Please help review to confirm if its okay now

**Pull request closed without merging**